### ISSUE TITLE
lldb fails to run properly on macOS

### ISSUE THREAD

**Issue by tromey** [rust-lang/rust#54126]
I installed the rust-enabled lldb from rustup and tried to run it.  It crashes with the mysterious:
```
(lldb) command script import "/Users/ttromey/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/etc/lldb_rust_formatters.py"
: CommandLine Error: Option 'limited-coverage-experimental' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
Assertion failed: (e == 0), function ~recursive_mutex, file /BuildRoot/Library/Caches/com.apple.xbs/Sources/libcxx/libcxx-400.9/src/mutex.cpp, line 64.
libc++abi.dylib: Pure virtual function called!
```
This seems to be a known LLVM bug: https://bugs.llvm.org/show_bug.cgi?id=22952.  See also https://bugs.llvm.org/show_bug.cgi?id=30587.  This is fallout from the switch to a dynamic LLVM, #53245.

I can't reproduce in my Linux build; not sure what is different there.

I'm looking into workarounds.  One option might be to hack our clang and lldb builds so that all libraries are marked `DISABLE_LLVM_LINK_LLVM_DYLIB`.

**Comment by tromey** [comment]
Can't reproduce with a local macOS build either.  I'm still investigating.

I wonder if it's possible to build clang and lldb out-of-tree as is done for lld.

**Comment by tromey** [comment]
If I run the rustup-installed lldb without going through `rust-lldb`, it works.

**Comment by tromey** [comment]
Aha, invoking the `command script import ...` command that `rust-lldb` provides is what causes the crash.  Which I guess could be apparent from the original failure.

**Comment by tromey** [comment]
Just a simple Python file with `import lldb` makes it crash, so my theory is that something is wrong with our lldb build, causing this to try to load some other liblldb.dylib.

**Comment by tromey** [comment]
The bug is that rustup does not preserve symlinks when installing lldb.

**Comment by tromey** [comment]
This is strange, though, because rustup and tar seem to have symlink logic.

**Comment by tromey** [comment]
The package manifest contains `file:.../_lldb.so`, so maybe this needs to be extended to allow a symlink somehow.

**Comment by tromey** [comment]
I think the answer ends up being that rustup-utils/src/utils.rs should understand how to copy a symlink.

**Comment by tromey** [comment]
I have a rustup patch I will test out tomorrow.

**Comment by tromey** [comment]
https://github.com/rust-lang-nursery/rustup.rs/pull/1504

**Comment by michaelwoerister** [comment]
Do you think this can be solved without going back to linking to LLVM statically?

**Comment by tromey** [comment]
Yes, see the PR -- once rustup can handle symlinks it should work.

I'm waiting for a new rustup release; then I'll test it again; and then if there aren't any more failures I'll write an announcement.

**Comment by michaelwoerister** [comment]
OK, great! So we don't need to revert that.

**Comment by vadimcn** [comment]
I wonder if [this](https://reviews.llvm.org/D52404) would fix it.

**Comment by tromey** [comment]
This is fixed by rustup 1.14.0, which shipped yesterday.
